lib/warren/handler/test.rb

```diff
@@ -3,7 +3,50 @@
 module Warren
   module Handler
     # Class Warren::Test provides provides a dummy RabbitMQ
-    # connection pool for use during testing
+    # connection pool for use during testing.
+    #
+    # = Set up a test warren
+    #
+    # By default, the test warren is disabled during testing to avoid storing
+    # messages unnecessarily. Instead you must explicitly enable it when you
+    # wish to test message receipt.
+    #
+    # If using rspec it is suggested that you add the following to your
+    # spec_helper.rb
+    #
+    #   config.around(:each, warren: true) do |ex|
+    #     Warren.handler.enable!
+    #     ex.run
+    #     Warren.handler.disable!
+    #   end
+    #
+    # = Making assertions
+    #
+    # It is possible to query the test warren about the messages it has seen.
+    # In particular the following methods are useful:
+    #
+    # {render:#messages}
+    #
+    # {render:#last_message}
+    #
+    # {render:#message_count}
+    #
+    # {render:#messages_matching}
+    #
+    # = Example
+    #
+    #   describe QcResult, warren: true do
+    #     let(:warren) { Warren.handler }
+    #
+    #     setup { warren.clear_messages }
+    #     let(:resource) { build :qc_result }
+    #     let(:routing_key) { 'test.message.qc_result.' }
+    #
+    #     it 'broadcasts the resource' do
+    #       resource.save!
+    #       expect(warren.messages_matching(routing_key)).to eq(1)
+    #     end
+    #   end
```

Comment:
These comments are very useful for anybody using the gem, as they probably would like to perform tests. Wouldn't it be better if they go in the main README?

Reply:
I'll pop a link to this in the readme, possibly with a brief outline. (Its all part of the Yard documentation, so is reasonably easy to access as a user of the gem.)